CompilerSource/backend/ideprint.h

```diff
@@ -16,30 +16,32 @@
 *** with this code. If not, see <http://www.gnu.org/licenses/>
 **/
 
+#ifndef ENIGMA_IDEPRINT_H
+#define ENIGMA_IDEPRINT_H
+
+#include "general/estring.h"
+
 #include <string>
 #include <cstdint>
+
 #define flushl '\n'
 // The MinGW team is the most thoughtless, incompetent bunch of inconsistent morons who ever lived
 
 struct ideprint {
   void (*f)(const char*);
-  ideprint &operator<< (std::string);
-  ideprint &operator<< (const char* x);
-  ideprint &operator<< (void* x);
-  ideprint &operator<< (int16_t x);
-  ideprint &operator<< (int32_t x);
-  ideprint &operator<< (int64_t x);
-  ideprint &operator<< (uint16_t x);
-  ideprint &operator<< (uint32_t x);
-  ideprint &operator<< (uint64_t x);
-  ideprint &operator<< (char x);
-  ideprint &operator<< (unsigned char x);
-  ideprint &operator<< (double x);
+  template<typename T> inline ideprint &operator<< (T x) { f(std::to_string(x).c_str()); return *this; }
   ideprint(void (*f)(const char*));
 };
 
+template<> inline ideprint& ideprint::operator<< <const char*>(const char* x) { f(x); return *this; }
+template<> inline ideprint& ideprint::operator<< <std::string>(std::string x) { f(x.c_str()); return *this; }
+template<> inline ideprint& ideprint::operator<< <void *>(void* x) { f(tostringv(x).c_str()); return *this; }
+template<> inline ideprint& ideprint::operator<< <char>(char x) { char a[2]; a[1] = 0; a[0] = x; f(a); return *this; }
```

Comment:
@JoshDreamland I tried to string(x, 1) here but ended up with garbage. Also I need the templates on this line or it won't compile.

Reply:
It's `string(1, x)`. But what you have is fine, I suppose (the buffer is stack-allocated). I might change it on principle. And yes, removing just `template<>` will make it not compile. You have to remove the `<std::string>`, `<void *>`, and `<char>` as well. But I guess compilers are okay with this syntax, now, given that nothing is screaming at you.

Reply:
I did remove both but I got duplicate declaration errors. This was only way that worked for me.

Reply:
I just noticed these declarations aren't in the struct. They would need to be for you to avoid link errors. Static-inline is default for functions implemented within a class. I'm now 0% convinced that this code should build in any compiler....

Reply:
well they build in 4 differenf versions of 2 compilers so...